lib/membrane_rtc_engine/engine.ex

```diff
@@ -2,17 +2,111 @@ defmodule Membrane.RTC.Engine do
   @moduledoc """
   RTC Engine implementation.
 
-  One RTC Engine instance is responsible for managing one room in which
-  all tracks of one peer are forwarded to all other peers.
+  RTC Engine is an abstraction layer responsible for linking together different types of `Endpoints`.
+  From implementation point of view RTC Engine is a `Membrane.Pipeline`.
+
+  ## Endpoints
+
+  `Endpoints` are `Membrane.Bin`s able to publish own tracks and subscribe for tracks from other Endpoints.
+  One can thought about Endpoint as an entity responsible for handling some specifc task.
+  Endpoint can be added and removed using `add_endpoint/3` and `remove_endpoint/1` respectively.
+
+  There are basically two types of Endpoints:
+  * Standalone Endpoints - they are in most cases spawned only once per RTC Engine instance and they are not associated with any peer.
+  * Peer Endpoints - they are associated with some peer.
```

Comment:
I will fix this but we have bidirectional relations between different terms (e.g. Messages <-> Media Events or Peers  <-> Endpoints) so it will be hard to avoid such situations anyway. In the new version this should be more redable.